620 clear map with other filters

mradamcox opened the pull request:
This PR implements a bit of a refactor and cleanup to the map code, along with closing #620. I got some [help with the geocoding control](https://github.com/maptiler/maptiler-geocoding-control/issues/96#issuecomment-3124721603) to achieve the clearing function, which comes down to this snippet:

```
gc.current.setOptions({apiKey:apiKey, clearOnBlur:true});
setTimeout(() => {
    gc.current.clearMap()
    gc.current.focus() 
    gc.current.blur() 
    gc.current.setOptions({apiKey:apiKey, clearOnBlur:false});
}, 1000);
```

However, I was having a lot of trouble figuring out how to expose the control instance in a way that these methods could actually be called in a way that affected the interface. What I ended up deciding to do was remove one layer of abstraction that we had been using, the `react-map-gl` library, and rely instead directly on `maplibre-gl`. (We were already using MapLibre, but accessing through the `react-map-gl` wrappers, and I think those React component wrappers were making it too hard to get to the level of control that we really needed--at least with my knowledge of React.)

So, in the end we now have one less dependency (yay!) and are only dealing with MapLibre classes directly, not mixing them with the React components that the other library provides on top of them. There are a few parts of the map code that are slightly more verbose now, but it's a net reduction in code, and overall I think it actually makes an easier-to-understand component.

After that was all handled, hooking up the control clearing to the "clear filters" button was straightforward, using a new property of the search.ui slice.

mradamcox commented:
> Do you think we should also reset the zoom level back to the full US view? Not necessary, just something to consider.

Yeah, that's a good idea, that will be an easy addition to make. I'll reset the map to the initial bounds that are used to create the map on load.

mradamcox commented:
I've also added a fix for part of #625, the problem with Buffalo, NY not showing up in the search control.

The deploy preview build failed the first time, but I think it was a netlify problem because I re-triggered it there and it has completed. Still seeing if I can re-trigger the checks that appear here on the pull request...

pengyin-shan commented:
> I've also added a fix for part of #625, the problem with Buffalo, NY not showing up in the search control.
> 
> The deploy preview build failed the first time, but I think it was a netlify problem because I re-triggered it there and it has completed. Still seeing if I can re-trigger the checks that appear here on the pull request...

Thank you @mradamcox ! Just that when testing this branch, I saw this after clicking the 'Clear All' button:
<img width="1846" height="848" alt="Screenshot 2025-07-29 at 10 26 10 AM" src="https://github.com/user-attachments/assets/63726df3-52ee-41cb-9b98-473f103abaa7" />

mradamcox commented:
The problem was this dependency array:

```
useEffect(() => {
        if (gcRef.current && !geosearchSelection) {
            gcRef.current.clearMap()
            mapRef.current.fitBounds(props.initialBounds)
            gcRef.current.setOptions({apiKey:apiKey, clearOnBlur:true});
            setTimeout(() => {
                gcRef.current.focus()
                gcRef.current.blur()
                gcRef.current.setOptions({apiKey:apiKey, clearOnBlur:false});
            }, 1000);
        }
   }, [geosearchSelection])
```

To please the linter I had added `props.initialBounds` to the dependency array just before commiting, but this causes the effect to loop and get to max recursion depth. For now, I've just removed it from the array and will live with the warning. May be something to look into later.

pengyin-shan reviewed (approved):
Tested both the "Clear All" feature and the location dropdown — everything works great. This PR will also help simplify future map operations. Approving the PR.
